client/src/cordova/dev.action.mjs

```diff
@@ -0,0 +1,142 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import os from 'os';
+import path from 'path';
+import url from 'url';
+
+import {createReloadServer} from '@outline/infrastructure/build/create_reload_server.mjs';
+import {getRootDir} from '@outline/infrastructure/build/get_root_dir.mjs';
+import {runAction} from '@outline/infrastructure/build/run_action.mjs';
+import {spawnStream} from '@outline/infrastructure/build/spawn_stream.mjs';
+import {hashElement} from 'folder-hash';
+import * as fs from 'fs-extra';
+import minimist from 'minimist';
+
+import {makeReplacements} from '../../build/make_replacements.mjs';
+
+const OUTPUT_PATH = 'output/build/client/macos';
+const OUTLINE_APP_PATH = 'Debug/Outline.app';
+const OUTLINE_APP_WWW_PATH = 'Contents/Resources/www';
+const RELOAD_SERVER_PORT = 35729;
+
+const getUIHash = async () => {
+  const hashResult = await hashElement(
+    path.join(getRootDir(), 'client/src/www'),
+    {
+      files: {include: ['**/*.ts', '**/*.html', '**/*.css', '**/*.js']},
+    }
+  );
+
+  return hashResult.hash;
+};
+
+/**
+ * @description Runs the cordova app in development mode.
+ *
+ * @param {string[]} parameters
+ */
+export async function main(...givenParameters) {
+  const {
+    _: [platform = 'macos'],
+    buildMode = 'release',
+    sentryDsn = 'https://public@sentry.example.com/1',
+    versionName = '0.0.0-dev',
+  } = minimist(givenParameters);
+
+  if (platform !== 'macos') {
+    throw new Error('This action currently only works for the MacOS platform.');
+  }
+
+  if (buildMode !== 'release') {
+    throw new Error('MacOS only renders properly in the release build mode.');
+  }
+
+  if (os.platform() !== 'darwin') {
+    throw new Error('You must be on MacOS to develop for MacOS.');
+  }
+
+  const parameters = [
+    'macos',
+    '--buildMode=release',
+    `--sentryDsn=${sentryDsn}`,
+    `--versionName=${versionName}`,
+  ];
+
+  await runAction('client/src/www/build', ...parameters);
+  await runAction('client/go/build', ...parameters);
+  await runAction('client/src/cordova/setup', ...parameters);
+
+  await makeReplacements([
+    {
+      files: path.join(
+        getRootDir(),
+        'client/platforms/osx/**/index_cordova.html'
+      ),
+      from: '<app-root></app-root>',
+      to: `<app-root></app-root>
+
+    <script>
+      try {
+        const reloadSocket = new WebSocket("ws://localhost:35729");
```

Comment:
Do you need to hard-code the server? Can this be a dynamic port of a unix socket instead?

You start the reload server, so you might as well use a dynamic port to avoid port conflicts.